README.md

```diff
@@ -1,18 +1,57 @@
-# {{.AppName}}
+# mongo-lock-node
 
-{{.Description}}
+Distributed lock client backed by mongo.
 
-Owned by {{.TeamName}}
+## Installation
+```
+#> npm install --save mongo-lock-node
+```
 
-## Developing
+## Usage
+```javascript
+import * as MongoClient from "mongodb";
+import {RWMutex} from "mongo-lock-node";
 
-- update package.json with your library name (`@clever/<name>` if private)
+// connect to mongo
+const db = await MongoClient.connect("mongodb://localhost:27017/test");
+const collection = db.collection("locks");
+await collection.createIndex("lockID", { unique: true });
 
-- `npm install`
+// create a client
+const lockID = "lock_1"; // must be unique per lock
+const clientID = "client_1"; // must be unique per client
+const lock = new RWMutex(collection, lockID, clientID);
 
-- Write the library in the `lib` folder
+// acquire the write lock
+try {
+  await lock.lock();
+} catch (err) {
+  // other try/catch blocks omitted for brevity
+  console.error(err.message); // => "error acquiring lock lock_1: connection interrupted"
+  throw err;
+}
+// do important stuff
+console.log("doing important things...");
+// release the write lock
+await lock.unlock();
+
+// acquire the read lock
+await lock.rLock();
+// read important things
+console.log("reading important things...");
```

Comment:
Might be nice to say here `Other things can also take the read lock`. Just a bit more detail about what having the read lock means